src/main/java/org/folio/consortia/service/impl/SharingRoleService.java

```diff
@@ -95,12 +95,14 @@ protected void deleteSharingConfig(UUID roleId) {
   }
 
   @Override
-  protected SharingRoleEntity createSharingConfigEntityFromRequest(SharingRoleRequest sharingRoleRequest, String tenantId) {
-    SharingRoleEntity sharingRoleEntity = new SharingRoleEntity();
-    sharingRoleEntity.setId(UUID.randomUUID());
-    sharingRoleEntity.setRoleId(sharingRoleEntity.getRoleId());
-    sharingRoleEntity.setTenantId(tenantId);
-    return sharingRoleEntity;
+  protected SharingRoleEntity createSharingConfigEntityFromRequest(SharingRoleRequest request, String tenantId) {
+    return SharingRoleEntity.builder()
+      .id(UUID.randomUUID())
+      .roleId(request.getRoleId())
```

Comment:
so this line was basically root cause of sharing issue, that roleId was not populated?

Reply:
yes  sharingRoleEntity.setRoleId(sharingRoleEntity.getRoleId()); roled is being populated using entity not request.

Reply:
![image](https://github.com/user-attachments/assets/07dad823-7da4-484b-b109-1048ce721367)